qiskit_machine_learning/kernels/quantum_kernel.py

```diff
@@ -47,13 +52,15 @@ class QuantumKernel:
     algorithms such as support vector classification, spectral clustering or ridge regression.
     """
 
+    @deprecate_arguments("0.5.1", {"user_parameters": "training_parameters"})
     def __init__(
         self,
         feature_map: Optional[QuantumCircuit] = None,
         enforce_psd: bool = True,
         batch_size: int = 900,
         quantum_instance: Optional[Union[QuantumInstance, Backend]] = None,
-        user_parameters: Optional[Union[ParameterVector, Sequence[Parameter]]] = None,
+        *,
+        training_parameters: Optional[Union[ParameterVector, Sequence[Parameter]]] = None,
```

Comment:
Is there a reason to force training_parameters to be a keyword parameter - it could still be used positionally where user_parameters was before right, if that's how someone called this, since its only a name change.

Reply:
two thoughts:
1. YES, you're right, and the code as-written almost-certainly has a bug: if user_parameters is supplied *without* a keyword-argument, the code simply won't work!  Oy, I overlooked this, and will fix.
2. My reasoning was: 
3. I found no examples of user_parameters being used as a positional argument.  And thought: "would be better to make training_parameters a keyword".  But sure, can retain it as positional.
4. *But*, once user_parameters is deprecated, training_parameters can return to being a possibly-positional argument.  Right now, it would be bad to allow users to write code that supplied both user_parameters and training_parameters as positional arguments -- because that would allow users to write code that would not work, once user_parameters was fully-removed.

Reply:
The idea of deprecation is that most importantly any working code they have written still works when they install the new version that introduced the deprecation and to give time for a user to migrate their code to the new way.

I agrees its unlikely someone has done all positional arguments but if they have their code should ideally continue to work. So far we have not forced users to do keyword arguments so I think I would leave it to still allow positional. If they have done positional they they will not notice the change in name. For sure someone should not be supplying both the old and newly named parameter - that's really a user error.